src/neural_mesh_simplification/trainer/trainer.py

```diff
@@ -122,36 +128,58 @@ def _train_one_epoch(self, epoch: int) -> float:
         running_loss = 0.0
         logger.debug(f"Starting epoch {epoch + 1}")
 
+        accumulation_steps = self.config["training"].get("accumulation_steps", 1)
+        self.optimizer.zero_grad()
+
         for batch_idx, batch in enumerate(self.train_loader):
             logger.debug(f"Processing batch {batch_idx + 1}")
             try:
-                self.optimizer.zero_grad()
+                batch = batch.to(self.device)
                 output = self.model(batch)
                 loss = self.criterion(batch, output)
 
                 del batch
+                del output
 
+                loss = loss / accumulation_steps  # Normalize the loss
                 loss.backward()
-                self.optimizer.step()
-                running_loss += loss.item()
+
+                if (batch_idx + 1) % accumulation_steps == 0:
+                    self.optimizer.step()
+                    self.optimizer.zero_grad()
+
+                running_loss += loss.item() * accumulation_steps
+
+                if (batch_idx + 1) % accumulation_steps == 0:
+                    logger.info(f"Batch {batch_idx + 1} - Loss: {running_loss / (batch_idx + 1):.4f}")
+
+                # Perform final optimization step if dataset size is not divisible by accumulation_steps
+                if (batch_idx + 1) % accumulation_steps != 0:
+                    self.optimizer.step()
+                    self.optimizer.zero_grad()
```

Comment:
Shouldn't this be outside the batch loop? To catch the remaining batches that wouldn't add up to the `accumulation_steps`?